bls-signatures-ffi/src/lib.rs

```diff
@@ -83,71 +97,69 @@ pub unsafe extern "C" fn aggregate(
 }
 
 /// Verify that a signature is the aggregated signature of hashes - pubkeys
-/// WARNING: This function can panic if there are a different number of digests and keys
 ///
 /// # Arguments
 ///
 /// * `signature_ptr`             - pointer to a signature byte array (SIGNATURE_BYTES long)
 /// * `flattened_digests_ptr`     - pointer to a byte array containing digests
 /// * `flattened_digests_len`     - length of the byte array (multiple of DIGEST_BYTES)
 /// * `flattened_public_keys_ptr` - pointer to a byte array containing public keys
-/// * `flattened_public_keys_len` - length of the byte array (multiple of PUBLIC_KEY_BYTES)
 #[no_mangle]
 pub unsafe extern "C" fn verify(
     signature_ptr: *const u8,
     flattened_digests_ptr: *const u8,
     flattened_digests_len: libc::size_t,
     flattened_public_keys_ptr: *const u8,
     flattened_public_keys_len: libc::size_t,
-) -> *mut responses::VerifyResponse {
```

Comment:
Observation: This is a breaking API change and will requite updates to `go-bls-sigs`. 

Rant: The response struct gives us a way to return an error message to callers (and, in other functions, a way to return multiple values). Establishing and following that convention (e.g. response structs with `err_message` and `err_code`) allows us to build tooling and makes consuming our FFI libs straightforward. I suggest that you stick with the convention and preserve the response struct.

Reply:
Usually I would agree, but for security reasons verification should not return anything else than true or false, as error messages could leak information

Reply:
I do not think that leaking information such as "flattened_digests_len != flattened_public_keys_len" constitutes a security vulnerability.

Reply:
it is a slippery slope, and much easier to get wrong than right, as one has to analyze each check